pandas/core/indexes/base.py

```diff
@@ -2490,12 +2490,10 @@ def _get_unique_index(self, dropna: bool = False):
         else:
             values = self._values
 
-        if dropna:
-            try:
-                if self.hasnans:
-                    values = values[~isna(values)]
-            except NotImplementedError:
-                pass
+        if dropna and not isinstance(self, ABCMultiIndex):
+            # isna not defined for MultiIndex
```

Comment:
It is reasonable to consider that ``ABCMultiIndex`` is never going to have ``isna`` defined?

Reply:
This came up on a dev call earlier this year and it didn't sound like there was reason for optimism.

Reply:
isna is tricky because a MI can be multi-valued e.g does isna refer to *any* value on a level, or all levels be null? it would be reasonable to actually return a multi-value here (e.g. same shape as the MI), but would need to see the implications if this is done.